### PR TITLE
Add docs for return types and options

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp [hunk 1]
```diff
@@ -153,7 +153,7 @@ namespace Azure { namespace Storage { namespace Blobs {
   };
 
   /**
-   * @brief Optional parameters for BlobServiceClient::ListBlobContainers.
+   * @brief Optional parameters for #Azure::Storage::Blobs::BlobServiceClient::ListBlobContainers.
```
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:156]: If the type is inside the same namespace we don't need to add the full namespace path, doxygen will resolve it.
However, that's a nit

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp [hunk 2]
```diff
@@ -29,54 +29,153 @@ namespace Azure { namespace Storage {
 
     namespace Models {
 
+      /**
+       * @brief Response type for #Azure::Storage::Blobs::BlobClient::DownloadTo.
+       */
       struct DownloadBlobToResult
       {
+        /**
+         * The blob's type.
```
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp:38]: =)

Comment (vhvb1989) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp:38]: maybe add a little info or at least a link to #Models::BlobType so people can click it to see the Type options

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp [hunk 3]
```diff
@@ -29,54 +29,153 @@ namespace Azure { namespace Storage {
...
+         */
         Models::BlobType BlobType;
+
+        /**
+         * Indicates the range of bytes returned.
```
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp:43]:
```suggestion
         * Indicates the #Azure::Core::Http::HttpRange of bytes returned.
```

**Review by vhvb1989 — COMMENTED** [review]
partial review.
This is one the perfect PRs for @ahsonkhan  =)

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:156]: I tried, it didn't work

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp (hunk 2 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp:38]: Yeah, totally we can. I just don't think the comment is necessary. The variable name is fairly self-explanatory.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp (hunk 3 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp:43]: The docs will look like

image
image

So it shows the type very clearly. I don't feel it's necessary to repeat it.

**Review by katmsft — APPROVED** [review]

**Pull request merged**